File: extensions/shared/src/main/java/app/morphe/extension/music/patches/ads/MusicAdsPatch.java
package app.morphe.extension.music.patches.ads;

import app.morphe.extension.music.settings.Settings;

@SuppressWarnings("unused")
public class MusicAdsPatch {
    private static final boolean HIDE_MUSIC_ADS = Settings.HIDE_MUSIC_ADS.get();

    public static boolean hideMusicAds() {
        return HIDE_MUSIC_ADS;
    }

    public static boolean hideMusicAds(boolean original) {
        return !HIDE_MUSIC_ADS && original;
    }
}

File: src/main/kotlin/app/revanced/patches/music/ads/general/MusicAdsPatch.kt
package app.revanced.patches.music.ads.general

import app.revanced.patches.music.utils.integrations.Constants.ADS_PATH
import app.revanced.patches.shared.ads.BaseAdsPatch

object MusicAdsPatch : BaseAdsPatch(
    "$ADS_PATH/MusicAdsPatch;",
    "hideMusicAds"
)
